src/components/GaugeWeight.tsx

```diff
@@ -0,0 +1,76 @@
+import React, { useRef } from "react"
+import { BigNumber } from "@ethersproject/bignumber"
+import { CircularProgress } from "@mui/material"
+import { Gauge } from "../providers/GaugeProvider"
+import Highcharts from "highcharts"
+import HighchartsExporting from "highcharts/modules/exporting"
+import HighchartsReact from "highcharts-react-official"
+import PieChart from "highcharts-react-official"
+
+interface Props {
+  gauges: Gauge[] | undefined
+}
+
+export default function GaugeWeight({
+  gauges,
+  ...props
+}: Props & HighchartsReact.Props): JSX.Element {
+  HighchartsExporting(Highcharts)
+  const chartComponentRef = useRef<HighchartsReact.RefObject>(null)
+  const data = gauges?.map((g) => {
```

Comment:
```
const basicPools = useContext(BasicPoolContext)
const gauges = useContext(GaugesContext)

const gaugeData = Object.values(basicPools).map(pool => {
  const gauge = gauges[pool.address] // or lpToken
  if (!gauge) return null
  return {poolName: pool.poolName, weight: gauge.relative_weight} // correct this
}).filter(Boolean)
```

Reply:
You can also generate this data in the GaugeVote component (the parent of GaugeWeight) and just pass in the data